### ISSUE TITLE
float as supported type for labels in classification scenarios

### ISSUE THREAD

**Issue by romanlutz** [fairlearn/fairlearn#95]
We should think about whether that's something we want to support, or whether we want to stick with int. If we don't want to support it we should at least check for the type and raise an exception with clear instructions for the user.

**Comment by paramrsingh** [comment]
This currently throws a TypeError in the first fit() method of the Binary Classification on COMPAS dataset notebook. Running v 0.3 of fairlearn on DSVM

**Comment by romanlutz** [comment]
I just ran this and I don't see a type error. Maybe you're running with an older version of tempeh? Try getting the latest fairlearn (0.4.0) and tempeh (0.1.11). @paramrsingh  let me know if you keep seeing the issue, and ideally also the version of tempeh. Thanks!

**Comment by adrinjalali** [comment]
Not sure if it's relevant here, but in sklearn this is what we do to [check the classification output](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/svm/_base.py#L523):

``` python
        y_ = column_or_1d(y, warn=True)
        check_classification_targets(y)
        cls, y = np.unique(y_, return_inverse=True)
```

**Comment by romanlutz** [comment]
It's mostly a question of error handling since there's no particularly good reason to support this. We should check for this and raise an Exception to let the user know that sensitive feature labels shouldn't be float (or if this is something we want then enable that to work).

**Comment by romanlutz** [comment]
To recap: we should add code to check whether the targets (`y`) are valid. We can use scikit-learn's `type_of_target`: https://github.com/scikit-learn/scikit-learn/blob/f2a6e109f7be6f0d554e44cb4cb48b41081dc259/sklearn/utils/multiclass.py#L201 
For mitigation techniques only binary classification and regression are currently supported. If `type_of_target` returns anything else we should raise an error.
This contribution should also have a set of tests to cover the various cases (binary and multi-class classification, regression, multi-label, etc.)
